force-app/main/default/classes/PlannedSalesTriggerHelper.cls

```diff
@@ -0,0 +1,138 @@
+/**
+ * @description       : 
+ * @author            : Nazar Voinarovych
+ * @group             : 
+ * @last modified on  : 10-07-2024
+ * @last modified by  : Nazar Voinarovych
+**/
+
+public with sharing class PlannedSalesTriggerHelper {
+    
+    public static void createPlannedSales(){
+        
+        List<Planned_Sales__c> createdPlannedSales=[SELECT Id,Year__c,Month__c,Car_Type__c,Total_Amount_Earned__c,Count_of_Cars_Sold__c 
+                                               FROM Planned_Sales__c WHERE Id IN:Trigger.New];
+        
+        //Map<Integer,Integer> monthAndYearOfPlannedSale=new Map<Integer,Integer>();
+
+        Set<Integer> month=new Set<Integer>();
+        Set<Integer> year=new Set<Integer>();
+        
+        
+        
+        Map<String,Planned_Sales__c> relatedPlannedSales=new Map<String,Planned_Sales__c>();
+        
+        Map<Id,Planned_Sales__c> plannedSalesToUpdate=new Map<Id,Planned_Sales__c>();
+                
+                
+        if(!createdPlannedSales.isEmpty()){
+            for(Planned_Sales__c plannedSale:createdPlannedSales){
+                String key=plannedSale.Month__c+'_'+plannedSale.Year__c+'_'+plannedSale.Car_Type__c;
+                relatedPlannedSales.put(key,plannedSale);
+                //! Problem HERE
+                month.add((Integer) plannedSale.Month__c);
+                year.add((Integer) plannedSale.Year__c);
+                //monthAndYearOfPlannedSale.put((Integer) plannedSale.Month__c,(Integer) plannedSale.Year__c);
+            }
+
+            if(month!=null && year!=null){
+                for(Opportunity opportunity:[SELECT Id,CloseDate,StageName,Amount,(SELECT Id,Product2.Condition__c FROM OpportunityLineItems) FROM Opportunity 
+                                              WHERE CALENDAR_MONTH(CloseDate) IN :month 
+                                             AND CALENDAR_YEAR(CloseDate) IN:year
+                                             AND StageName='Closed Won'
+                                            ]){ 
+                     
+                    String key=opportunity.CloseDate.month()+'_'+opportunity.CloseDate.year(); 
+                                                 
+                    for(OpportunityLineItem oli:opportunity.OpportunityLineItems){
+                        
+                        String key2=key+'_'+oli.Product2.Condition__c;
+                        
+                        if(relatedPlannedSales.containsKey(key2)){
+                            
+                            Planned_Sales__c plannedSale=relatedPlannedSales.get(key2);
+                            
+                            if(plannedSale!=null){
+                                plannedSale.Total_Amount_Earned__c+=opportunity.Amount;
+                                plannedSale.Count_of_Cars_Sold__c+=opportunity.OpportunityLineItems.size();
```

Comment:
Adding Opportunity line item amount on each LineItem iteration will not calculate if correctly

If you have 10 cars each 100k -> Opp.Amount  = 1 mil, 10 iterations will add 10 mil instead of 1 min

Another thing with Count_of_Cars_Sold__c. You have to only add cars with same Condition as Car Type on planned sales